### PR TITLE
Add a reason field to report command

### PR DESCRIPTION
**Pull request by 3raphat**
# Description

Able to specify the reason for the report

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshot

image
image

# Checklist:

- [x] I have run `pnpm format` and my code don't have any linting issues
  
  
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/commands/moderators/report.ts [hunk 1]
```diff
@@ -22,24 +30,62 @@ export default defineCommandHandler({
...
+      new ActionRowBuilder<ModalActionRowComponentBuilder>().addComponents(
+        reasonInput,
+      ),
+    )
+
+    // Show the modal
+    await interaction.showModal(reasonModal)
+
+    const submitted = await interaction.awaitModalSubmit({
+      filter: (interaction) => interaction.user.id === interaction.user.id,
```
Comment (dtinth) [inline src/commands/moderators/report.ts:53]: this condition is always true, maybe `interaction_`?

Comment (3raphat) [reply src/commands/moderators/report.ts:53]:
```suggestion
      filter: (interaction_) => interaction_.user.id === interaction.user.id,
```
like this? @dtinth

Comment (dtinth) [reply src/commands/moderators/report.ts:53]: yes krub!

**Comment by beam41** [comment]
Code is good, but you should limit max length for reason text field (someone might abuse report with that 4000 character length). I think 50 is enough.

**Comment by 3raphat** [comment]
> Code is good, but you should limit max length for reason text field (someone might abuse report with that 4000 character length). I think 50 is enough.

I will set it to 100 for someone who wants to explain the reason.

ป.ล. ภาษาไทยมีสระ วรรณยุกต์กินที่เยอะกว่าอังกฤษด้วย

**Review by beam41 — APPROVED** [review]

**Review by dtinth — APPROVED** [review]
LGTM — let’s merge!

**Pull request merged**